auto_dev/data/templates/protocols/primitives.jinja

```diff
@@ -0,0 +1,158 @@
+import struct
+from abc import ABC, abstractmethod
+from pydantic_core import SchemaValidator, core_schema
+
+
+min_int32 = -1 << 31
+max_int32 = (1 << 31) - 1
+min_uint32 = 0
+max_uint32 = (1 << 32) - 1
+
+min_int64 = -1 << 63
+max_int64 = (1 << 63) - 1
+min_uint64 = 0
+max_uint64 = (1 << 64) - 1
+
+min_float32 = struct.unpack('f', struct.pack('I', 0xFF7FFFFF))[0]
+max_float32 = struct.unpack('f', struct.pack('I', 0x7F7FFFFF))[0]
+min_float64 = struct.unpack('d', struct.pack('Q', 0xFFEFFFFFFFFFFFFF))[0]
+max_float64 = struct.unpack('d', struct.pack('Q', 0x7FEFFFFFFFFFFFFF))[0]
+
+
+class BaseConstrainedFloat(float, ABC):
+    """Base class for constrained float types."""
+
+    @classmethod
+    @abstractmethod
+    def min(cls) -> float:
+        raise NotImplementedError(f"{cls.__name__}.min() is not implemented.")
+
+    @classmethod
+    @abstractmethod
+    def max(cls) -> float:
+        raise NotImplementedError(f"{cls.__name__}.max() is not implemented.")
+
+    def __new__(cls, value: float = 0.0, *args, **kwargs) -> "BaseConstrainedFloat":
+        schema = core_schema.float_schema(strict=True, ge=cls.min(), le=cls.max())
+        validator = SchemaValidator(schema)
+        validated_value = validator.validate_python(value)
+        return super().__new__(cls, validated_value)
+
+    def __new__(cls, value: float = 0.0, *args, **kwargs) -> "BaseConstrainedInt":
+        schema = core_schema.float_schema(strict=True, ge=cls.min(), le=cls.max())
+        validator = SchemaValidator(schema)
+        validated_value = validator.validate_python(value)
+        return super().__new__(cls, validated_value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source, handler):
+        schema = core_schema.float_schema(strict=True, ge=cls.min(), le=cls.max())
+        return core_schema.no_info_wrap_validator_function(cls, schema)
```

Comment:
Custom type system ensures proper encoding to `int` and `float` for the correct `protobuf` type, with validation on initialization. The `__get_pydantic_core_schema__` method is necessary to integrate the custom constraints (min/max) with Pydantic’s validation system, ensuring correct validation and serialization behavior.